spec/integration/type/tidy_spec.rb

```diff
@@ -11,6 +11,28 @@
     Puppet::Util::Storage.stubs(:store)
   end
 
+  it "should be able to recursively remove directories" do
+    dir = tmpfile("tidy_testing")
+    foo = File.join(dir, "foo")
+    bar = File.join(foo, "bar")
+
+    Dir.mkdir(dir)
+    Dir.mkdir(foo)
+    Dir.mkdir(bar)
+
+    tidy = Puppet::Type.type(:tidy).new :path => dir, :recurse => true, :rmdirs => true
+
+    catalog = Puppet::Resource::Catalog.new
+    catalog.add_resource(tidy)
+    # avoid crude failures because of nil resources that result
+    # from implicit containment and lacking containers
+    catalog.stubs(:container_of).returns tidy
+
+    catalog.apply
+
+    expect(Puppet::FileSystem.directory?(dir)).to be_falsey
```

Comment:
So it fails on a line above, and I guess I could change it to an "expect the apply to not raise", but this is more what the test ought to look like for testing the actual functionality (vs testing the bug that was introduced)